### PR TITLE
fix(package): Normalize tables when writing tables.json

### PR DESCRIPTION
**Pull request by erezrokah**
#### Summary

We might want to refactor this to no (ab)use the MemDB plugin but I think this is ok for now. Also we should call `FlattenTables` otherwise we only output the top level ones

---

Use the following steps to ensure your PR is ready to be reviewed

- [ ] Read the [contribution guidelines](../blob/main/CONTRIBUTING.md) 🧑‍🎓
- [ ] Run `go fmt` to format your code 🖊
- [ ] Lint your changes via `golangci-lint run` 🚨 (install golangci-lint [here](https://golangci-lint.run/usage/install/#local-installation))
- [ ] Update or add tests 🧪
- [ ] Ensure the status checks below are successful ✅

### CODE REVIEW

File: serve/package.go [hunk 1]
```diff
@@ -46,18 +46,49 @@ type TargetBuild struct {
...
+			Name:          table.Name,
+			Parent:        parent,
+			Relations:     relations,
+			Title:         table.Title,
+		})
+	}
 	buffer := &bytes.Buffer{}
 	m := json.NewEncoder(buffer)
-	m.SetIndent("", "  ")
+	m.SetIndent("", "")
```
Comment (erezrokah) [inline serve/package.go:89]: I think we want this JSON as compact as possible

File: serve/plugin_test.go [hunk 2]
```diff
@@ -80,7 +80,7 @@ func TestPluginServe(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	if len(tables) != 0 {
+	if len(tables) != 2 {
 		t.Fatalf("Expected 0 tables but got %d", len(tables))
```
Comment (hermanschaaf) [inline serve/plugin_test.go:84]:
```suggestion
		t.Fatalf("Expected 2 tables but got %d", len(tables))
```

File: serve/package_test.go [hunk 3]
```diff
@@ -125,6 +126,18 @@ func checkDocs(t *testing.T, dir string, expect []string) {
...
 
+func checkTables(t *testing.T, distDir string) {
+	content, err := os.ReadFile(filepath.Join(distDir, "tables.json"))
+	if err != nil {
+		t.Fatal(err)
+	}
+	tablesString := string(content)
+
+	if diff := cmp.Diff(tablesString, "[{\"name\":\"table1\",\"relations\":[\"table2\"]},{\"name\":\"table2\"}]\n"); diff != "" {
+		t.Fatalf("unexpected files in docs directory (-want +got):\n%s", diff)
```
Comment (hermanschaaf) [inline serve/package_test.go:137]:
```suggestion
		t.Fatalf("unexpected content in tables.json (-want +got):\n%s", diff)
```

**Review by hermanschaaf — APPROVED** [review]

File: serve/package.go [hunk 4]
```diff
@@ -46,18 +46,50 @@ type TargetBuild struct {
...
 }
 
+// This is the structure the CLI publish command expects
+type pluginTable struct {
+	Description   string    `json:"description,omitempty"`
+	IsIncremental bool      `json:"is_incremental,omitempty"`
+	Name          string    `json:"name,omitempty"`
+	Parent        *string   `json:"parent,omitempty"`
+	Relations     *[]string `json:"relations,omitempty"`
+	Title         string    `json:"title,omitempty"`
```
Comment (hermanschaaf) [inline serve/package.go:56]: Don't we also need the columns?

**Pull request merged**